mage_ai/data_cleaner/cleaning_rules/remove_collinear_columns.py

```diff
@@ -20,6 +20,7 @@ def evaluate(self):
         if self.numeric_df.empty or len(self.numeric_df) < self.MIN_ENTRIES:
             return suggestions
         collinear_columns = []
+        self.numeric_df['intercept'] = np.ones(len(self.numeric_df))
```

Comment:
Intercept will not be included in the suggestion. The only columns that are checked for collinearity are the numerical columns stored in `self.numeric_columns` as found by `filter_numeric_types()` on line 15. This list was generated separately and has no references to `self.numeric_df.columns`. As a result, when intercept is only added on line 23, it will not be added to `self.numeric_columns` and will not be checked for collinearity and reported in a suggestion.

Additionally, if there are no numeric features to begin with, an intercept will never be added so a suggestion cannot be made.